src/components/buoy-telemetry/BuoyMap.vue

```diff
@@ -0,0 +1,155 @@
+<template>
+  <div class="mapboxgl-map-container">
+    <div class="date is-hidden-touch">
+      <span>{{ store.maxDate }}</span>
+    </div>
+    <div class="date-mobile is-hidden-desktop">
+      <span>{{ store.maxDate }}</span>
+    </div>
+    <div ref="el" class="mapbox-container" />
+  </div>
+
+  <!-- reference image for mapbox to pull from -->
+  <img
+    ref="imageEl"
+    :src="BuoyMarker"
+    style="
+      visibility: hidden;
+      height: 15px;
+      width: 15px;
+      position: absolute;
+      top: 0;
+    "
+  />
+</template>
+
+<script setup lang="ts">
+import { inject, onMounted, ref } from "vue";
+import mapboxgl from "mapbox-gl";
+
+import BuoyMarker from "@/assets/illustrations/buoy-marker.svg";
+
+import { BuoyStore } from "../../store/buoy";
+const store = inject("store") as BuoyStore;
+
+const el = ref<HTMLDivElement>(null);
+const imageEl = ref<HTMLImageElement>(null);
+const map = ref(null);
+onMounted(() => {
+  mapboxgl.accessToken = import.meta.env.VITE_MAPBOX_ACCESS_TOKEN;
+
+  map.value = new mapboxgl.Map({
+    container: el.value,
+    style: "mapbox://styles/ccv-bot/ckmxra8oi0rsw17mzcbqrktzi",
+    center: [-71.39, 41.6],
+    zoom: 9.9,
+    doubleClickZoom: false,
+    scrollZoom: false,
+  });
+
+  const geoJSON = {
+    type: "geojson",
+    data: {
+      id: "buoy",
+      type: "FeatureCollection",
+      features: store.coordinates.map(({ latitude, longitude }) => {
+        return {
+          type: "Feature",
+          geometry: {
+            type: "Point",
+            coordinates: [longitude, latitude],
+          },
+        };
+      }),
+    },
+  };
+
+  map.value.on("load", function () {
+    map.value.resize();
+    map.value.addImage("buoy-marker", imageEl.value);
+
+    map.value.addSource("points", geoJSON);
+
+    map.value.addLayer({
+      id: "points",
+      type: "symbol",
+      source: "points",
+      layout: {
+        "icon-allow-overlap": true,
+        "icon-image": "buoy-marker",
+        "icon-offset": [0.5, -7], // optically centered
+      },
+    });
+
+    updateMarkers();
+  });
+});
+
+let markers = [];
+const updateMarkers = () => {
+  markers.forEach((marker) => {
+    marker._element.remove();
+    marker.remove();
+  });
+  markers = [];
+
+  store.selectedSamples.forEach(({ color, size, station_name }) => {
+    // create circle elements to display
+    const elementSize = `${100 * size}px`;
+    const el = document.createElement("div");
+    el.className = "marker";
+    el.style.width = elementSize;
+    el.style.height = elementSize;
+    el.style.backgroundColor = color;
+    el.style.opacity = "0.8";
+    el.style.borderRadius = "100%";
+
+    markers.push(
+      new mapboxgl.Marker({ element: el })
+        .setLngLat(store.siteCoordinates(station_name))
+        .setPopup(new mapboxgl.Popup().setText(station_name))
+        .addTo(map.value)
+    );
+  });
+};
+</script>
+
+<style lang="scss" scoped>
```

Comment:
These styles are the same as in `da/BuoyMap.vue`. Is there a more centralized place we could put them?

Reply:
good idea, I just need to figure out where

Reply:
Did this get figured out?